### PR TITLE
Add new exception report notebook for COS

### PR DESCRIPTION
**Pull request by srosagomez**
This notebook checklist has been made available to us by the [Notebooks For All team](https://github.com/Iota-School/notebooks-for-all/blob/main/resources/event-hackathon/notebook-authoring-checklist.md).
Its purpose is to serve as a guide for both the notebook author and the technical reviewer highlighting critical aspects to consider when striving to develop an accessible and effective notebook.

### The First Cell

- [ ] The title of the notebook in a first-level heading (eg. `<h1>` or `# in markdown`).
- [ ] A brief description of the notebook.
- [ ] A table of contents in an [ordered list](https://www.markdownguide.org/basic-syntax/#ordered-lists) (`1., 2.,` etc. in Markdown).
- [ ] The author(s) and affiliation(s) (if relevant).
- [ ] The date first published.
- [ ] The date last edited (if relevant).
- [ ] A link to the notebook's source(s) (if relevant).

### The Rest of the Cells

- [ ] There is only one H1 (`#` in Markdown) used in the notebook.
- [ ] The notebook uses other [heading tags](https://www.markdownguide.org/basic-syntax/#headings) in order (meaning it does not skip numbers). 

## Text

- [ ] All link text is descriptive. It tells users where they will be taken if they open the link.
- [ ] All acronyms are defined at least the first time they are used. 
- [ ] Field-specific/specialized terms are used when needed, but not excessively.

## Code

- [ ] Code sections are introduced and explained before they appear in the notebook. This can be fulfilled with a heading in a prior Markdown cell, a sentence preceding it, or a code comment in the code section.
- [ ] Code has explanatory comments (if relevant). This is most important for long sections of code.
- [ ] If the author has control over the syntax highlighting theme in the notebook, that theme has enough color contrast to be legible.
- [ ] Code and code explanations focus on one task at a time. Unless comparison is the point of the notebook, only one method for completing the task is described at a time.

### Images

- [ ] All images (jpg, png, svgs) have an [image description](https://www.w3.org/WAI/tutorials/images/decision-tree/). This could be
    - [ ] [Alt text](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/alt) (an `alt` property) 
    - [ ] [Empty alt text](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/alt#decorative_images) for decorative images/images meant to be skipped (an `alt` attribute with no value)
    - [ ] Captions
    - [ ] If no other options will work, the image is decribed in surrounding paragraphs.

- [ ] Any [text present in images](https://www.w3.org/WAI/WCAG21/Understanding/images-of-text.html) exists in a text form outside of the image (this can be alt text, captions, or surrounding text.)

### Visualizations

- [ ] All visualizations have an image description. Review the previous section, Images, for more information on how to add it.
- [ ] [Visualization descriptions](http://diagramcenter.org/specific-guidelines-e.html) include
    - [ ] The type of visualization (like bar chart, scatter plot, etc.)
    - [ ] Title
    - [ ] Axis labels and range
    - [ ] Key or legend
    - [ ] An explanation of the visualization's significance to the notebook (like the trend, an outlier in the data, what the author learned from it, etc.)

- [ ] All visualizations and their parts have [enough color contrast](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html) ([color contrast checker](https://webaim.org/resources/contrastchecker/)) to be legible. Remember that transparent colors have lower contrast than their opaque versions.
- [ ] All visualizations [convey information with more visual cues than color coding](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html). Use text labels, patterns, or icons alongside color to achieve this.
- [ ] All visualizations have an additional way for notebook readers to access the information. Linking to the original data, including a table of the data in the same notebook, or sonifying the plot are all options.

### CODE REVIEW

File: notebooks/COS/ExceptionReport/cos_functions.py [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+        if verbose:
+            print(
+                "\nThis range was found on multiple segments, (grating = G230L?) ,which at present is not fully supported. The returned array should be accurate, but the mean may be incorrect."
+            )
+    return weight_avg_snr, snr_array
+
+
+# %%
+def withinPercent(val1, val2, percent=1.0):
+    """
```
Comment (haticekaratay) [inline notebooks/COS/ExceptionReport/cos_functions.py:293]: I don't see this being called anywhere. Could you please add instructions on how to test in the notebook or consider removing it entirely?

File: notebooks/COS/ExceptionReport/cos_functions.py [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+    percentDif = np.abs(100 * (absDif / lowval))
+    within_percent_bool = percentDif <= percent
+    return within_percent_bool, percentDif
+
+
+# %%
+"""
+This section is for testing and validating our S/N Calculations.
+"""
+if __name__ == "__main__":
```
Comment (haticekaratay) [inline notebooks/COS/ExceptionReport/cos_functions.py:317]: I was reviewing the script and noticed the `if __name__ == "__main__":` block used for testing. I just wanted to flag that this might not be immediately clear to many users, especially those working primarily in notebooks. Typically, notebook users won't be running scripts directly from the command line, and this main block might not be the best fit for those workflows. 
- If you want users to be able to run this test from their notebooks, it might be helpful to provide some instructions on how to do so. Notebook users may not know how to interact with the main block, and it could create confusion. OR
- You could move the logic in the main block into a dedicated function. This way, users can call the function directly from their notebook or another script without needing to figure out how to trigger the main() block. It will also allow you to run tests with a simple function call rather than hardcoding things.
-  also noticed that the file path in the main() block is hardcoded, but it seems like the file doesn’t exist. It might be better to either check for the file’s existence or make the path dynamic (e.g., allow the user to input the file path).
In short :)
If testing isn’t needed in the notebook workflow, consider removing the `if __name__ == "__main__":` block entirely.
If you want users to be able to test from the notebook, encapsulate the logic into a separate function with clear instructions.
Let me know what you think, and feel free to reach out if you'd like to discuss this further. Thanks again for all your hard work on this!

File: notebooks/COS/ExceptionReport/ExceptionReport.ipynb [hunk 3]
```diff
@@ -0,0 +1,1448 @@
+{
```
Comment (haticekaratay) [inline notebooks/COS/ExceptionReport/ExceptionReport.ipynb:1]: <p>There is some code duplication in plotting the same data tice for flux. vs wavelength. You can wrap the plotting logic into a loop to reduce redundancy.</p><p> </p>

<pre class="ql-syntax" spellcheck="false">for i, axis in enumerate(ax[0]):
&nbsp; &nbsp; # Plotting flux vs wavelength on top row
&nbsp; &nbsp; axis.plot(wl, flux,
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; lw=2,
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; color="black")
&nbsp; &nbsp; axis.set_xlabel(r'Wavelength [$\AA$]')
&nbsp; &nbsp; axis.set_ylabel(r'Flux [$erg\ s^{-1}\ cm^{-2}\ \AA^{-1}$]')
&nbsp; &nbsp; if i == 0:
&nbsp; &nbsp; &nbsp; &nbsp; axis.set_title("Flux vs Wavelength")
&nbsp; &nbsp; else:
&nbsp; &nbsp; &nbsp; &nbsp; axis.set_title("Flux vs Wavelength, ZOOMED")
&nbsp; &nbsp; &nbsp; &nbsp; axis.set_ylim(0, 0.6e-13)


</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/hst_notebooks/pull/305/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (haticekaratay) [inline notebooks/COS/ExceptionReport/ExceptionReport.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #12.    
</pre><p> </p><p>Please follow the suggestion given in the example above.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/hst_notebooks/pull/305/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

**Comment by haticekaratay** [comment]
Hi @srosagomez,
I liked your notebook—it’s well-written and easy to follow. There’s not much that needs updating. I’ve added a few minor comments above, but that’s it!

Great job, and thanks for your work on this!

File: notebooks/COS/ExceptionReport/ExceptionReport.ipynb (hunk 3 above)
Comment (haticekaratay) [reply notebooks/COS/ExceptionReport/ExceptionReport.ipynb:1]: @srosagomez, this feedback is from an engineering perspective. However, if you prefer to keep your tutorial as is for clarity, that's fine with me. A clear workflow is more important than avoiding code duplication in the tutorial. Please check the readability and let me know :)

**Comment by haticekaratay** [comment]
This is a new notebook. Could you also please update the toc.yml so that it renders on the HTML pages?

File: notebooks/COS/ExceptionReport/ExceptionReport.ipynb (hunk 3 above)
Comment (haticekaratay) [reply notebooks/COS/ExceptionReport/ExceptionReport.ipynb:1]: <p>I noticed that you have already implemented the code in the above cell. Could you please do the same for this code cell?</p>

**Review by haticekaratay — APPROVED** [review]
Thanks for your work! All looks good to me!
